### DISCUSSION TITLE
Is slipcover going to be a drop-in replacement for coverage.py?

### DISCUSSION THREAD

**Discussion by anilbey** [plasma-umass/slipcover#20] [General]
Hi everyone,

Thanks for building slipcover!
I think it would be informative to clarify if slipcover is going to be a drop-in replacement for coverage.py.

If slipcover can provide the same output files (in the same format) as coverage.py, I think it will be very easy for users to try slipcover in their CI systems.
E.g. third party apps like codecov (https://github.com/marketplace/codecov) can be configured without effort.

What do you think?

**Comment by jaltmayerpizzorno** [comment]
Hi!  To the extent our functionality matches coverage.py, we've attempted to stay compatible.
Is there anything in particular (format/options/...) that you'd like to see in Slipcover?

**Reply by anilbey** [reply]
It would be great to use slipcover with the `pytest-cov` package.

https://pypi.org/project/pytest-cov/

**Reply by anilbey** [reply]
I am not totally sure but using `pytest-cov` may also solve this (https://github.com/plasma-umass/slipcover/issues/12) issue, since it has this `--cov-append` option to combine multiple coverage files.

**Reply by wsascha** [reply]
Does `slipcover` have options to create the same coverage output files as `coverage.py`? Like XML, JSON, (maybe even html?) :-)

**Reply by jaltmayerpizzorno** [reply]
It has JSON.

**Comment by roniemartinez** [comment]
Another tool is SonarCloud which only supports XML for Python. This prevents usage of Slipcover as a replacement/alternative to coverage.py.

Kudos BTW!
